SwiftKitCore/src/main/java/org/swift/swiftkitcore/JNISwiftInstanceCleanup.java

```diff
@@ -0,0 +1,28 @@
+package org.swift.swiftkitcore;
+
+import java.lang.reflect.Method;
+
+class JNISwiftInstanceCleanup implements SwiftInstanceCleanup {
+    private final Class<? extends SwiftInstance> clazz;
+    private final long selfPointer;
+    private final Runnable markAsDestroyed;
+
+    public JNISwiftInstanceCleanup(Class<? extends SwiftInstance> clazz, long selfPointer, Runnable markAsDestroyed) {
+        this.clazz = clazz;
+        this.selfPointer = selfPointer;
+        this.markAsDestroyed = markAsDestroyed;
+    }
+
+    @Override
+    public void run() {
+        markAsDestroyed.run();
+
+        try {
+            // Use reflection to look for the static destroy method on the wrapper.
+            Method method = clazz.getDeclaredMethod("destroy", long.class);
```

Comment:
We can pass a Function rather than `Class` to the constructor and store that; We shouldn't need to resort to reflection here AFAICS.